src/RewardsDistribution.t.sol

```diff
@@ -0,0 +1,253 @@
+
+        vm.expectRevert("RewardsDistribution/invalid-vest-id");
+        dist.distribute();
+    }
+
+    function testRevertDistributeNoVestedAmount() public {
+        vm.expectRevert("RewardsDistribution/empty-vest");
+        dist.distribute();
+    }
+
+    function testRevertDistributeNoPendingDistributionAmount() public {
+        dist.file("calc", address(new MockZeroDistribution()));
+
+        skip(duration / 3);
+
+        vm.expectRevert("RewardsDistribution/no-pending-amount");
+        dist.distribute();
+    }
+
+    function testRevertFileInvalidVestId() public {
+        vm.expectRevert("RewardsDistribution/invalid-vest-id");
+        dist.file("vestId", 100);
+    }
+
+    function testRevertFileInvalidUsr() public {
+        address usr = address(0x1337);
+        uint256 tot = totalRewards;
+        uint256 bgn = block.timestamp; // start immediately
+        uint256 tau = duration; // 1 year duration
+        uint256 eta = 0; // No cliff; start immediatebly
+        address mgr = address(dist);
+        uint256 newVestId = _createVest(usr, tot, bgn, tau, eta, mgr);
+
+        vm.expectRevert("RewardsDistribution/invalid-vest-usr");
+        dist.file("vestId", newVestId);
+    }
+
+    function testRevertFileInvalidMgr() public {
+        address usr = address(farm);
+        uint256 tot = totalRewards;
+        uint256 bgn = block.timestamp; // start immediately
+        uint256 tau = duration; // 1 year duration
+        uint256 eta = 0; // No cliff; start immediatebly
+        address mgr = address(0x1337);
+        uint256 newVestId = _createVest(usr, tot, bgn, tau, eta, mgr);
+
+        vm.expectRevert("RewardsDistribution/invalid-vest-mgr");
+        dist.file("vestId", newVestId);
+    }
+
+    function testAuth() public {
+        checkAuth(address(dist), "RewardsDistribution");
+    }
+
+    function testFile() public {
+        // `checkFileUint` increaments the current value of the param being modified. Since `vestId` is validated,
+        // we need to create a new one to make sure the `file`d param is valid.
+        _createVest();
+        checkFileUint(address(dist), "RewardsDistribution", ["vestId"]);
+
+        checkFileAddress(address(dist), "RewardsDistribution", ["calc"]);
+    }
+
+    function testModifiers(address sender) private {
+        vm.assume(sender != address(this));
+
+        bytes4[] memory authedMethods = new bytes4[](1);
+        // authedMethods[0] = RewardsDistribution.mint.selector;
+
+        vm.startPrank(sender);
+        checkModifier(address(dist), "RewardsDistribution/not-authorized", authedMethods);
+    }
+
+    function _createVest() internal returns (uint256 _vestId) {
+        address usr = address(farm);
+        uint256 tot = totalRewards;
+        uint256 bgn = block.timestamp; // start immediately
+        uint256 tau = duration; // 1 year duration
+        uint256 eta = 0; // No cliff; start immediatebly
+        address mgr = address(dist);
```

Comment:
`mgr` is just to `yank()` so this can be set to `address(0)`

Reply:
Thanks for pointing that out, I somehow overlooked this and assumed something completely different.
It's fixed now.

Reply:
Fixed. Can't resolve comments.